test/unit/specs/core/createEventManager.spec.js

```diff
@@ -108,31 +108,51 @@ describe("createEventManager", () => {
         });
     });
 
-    it("sets the lastChanceCallback, which wraps config.onBeforeEventSend, on the event", () => {
-      let wrappedLastChanceCallback;
-      event.setLastChanceCallback.and.callFake(callback => {
-        wrappedLastChanceCallback = callback;
+    it("events call onBeforeEventSend callback", () => {
+      event.finalize.and.callFake(onBeforeEventSend => {
+        onBeforeEventSend();
       });
-      return eventManager.sendEvent(event, {}).then(() => {
-        wrappedLastChanceCallback();
+      return eventManager.sendEvent(event).then(() => {
         expect(config.onBeforeEventSend).toHaveBeenCalled();
       });
     });
 
-    it("logs errors in the config.onBeforeEventSend callback", () => {
-      const error = new Error("onBeforeEventSend error");
-      config.onBeforeEventSend.and.throwError(error);
+    it("returns false onBeforeEventSend and event should not be sent", () => {
+      const eventToIgnore = jasmine.createSpyObj("event", {
+        finalize: () => {},
+        shouldSend: false
+      });
+      return eventManager.sendEvent(eventToIgnore).then(() => {
+        expect(sendEdgeNetworkRequest).not.toHaveBeenCalled();
+      });
+    });
 
-      let wrappedLastChanceCallback;
-      event.setLastChanceCallback.and.callFake(callback => {
-        wrappedLastChanceCallback = callback;
+    it("returns null onBeforeEventSend and event should be sent", () => {
```

Comment:
I *think* what you're saying is that if `onBeforeEventSend` returns null, the event should still be sent. I don't think that's what's really being tested here though. This test is currently testing that if `shouldSend` returns null, that the event will still be sent, but I think this is an invalid test case because `shouldSend` always returns a boolean, as far as I can tell.

I believe what we need to do is remove this test case  and instead make sure that `createEvent.spec.js` has a test that asserts that when `onBeforeEventSend` returns `null`, that the event then returns `true` from `shouldSend`.